include/cling/Interpreter/Interpreter.h

```diff
@@ -12,6 +12,8 @@
 
 #include "cling/Interpreter/InvocationOptions.h"
 
+#include "llvm/ADT/StringRef.h"
```

Comment:
This is intentional. Cling's DynamicLookup `#includes "Interpreter.h"` at runtime. This means that systems like ROOT, need to provide all headers that are included in the Interpreter.h and the indirect includes thereof.

StringRef has been very fluctuating over the upgrades. IIRC, now it includes stuff outside `llvm/ADT` (llvm/Support/STLExtras.h, IIRC). This means that we need even more scaffolding to move. This doesn't simplify the overly complicated llvm upgrade procedure.

That said, @Axel-Naumann is planning to try to remove the dependency on the Interpreter.h in cling's runtime. This would hopefully allow Interpreter.h to be able to include again things like StringRef.